### PR TITLE
applications: serial_lte_modem: Fix build with native TLS

### PR DESCRIPTION
**Pull request by rlubos**
Adjust the sample with nrf_security configuration changes.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

**Comment by rlubos** [comment]
CC @SebastianBoe

**Comment by SebastianBoe** [comment]
Waiting for review by Frank

**Review by SebastianBoe — CHANGES_REQUESTED** [review]
A different approach is needed.

File: applications/serial_lte_modem/overlay-native_tls.conf
Comment (frkv) [inline applications/serial_lte_modem/overlay-native_tls.conf]: No longer needed (Will not have any effect)

File: applications/serial_lte_modem/overlay-native_tls.conf
Comment (frkv) [inline applications/serial_lte_modem/overlay-native_tls.conf]: Fixed in subsequent update to nrfxlib, no longer needed

**Comment by SebastianBoe** [comment]
@frkv : are the things you are not commenting on still needed or should this be dropped in it's entirety?

**Pull request merged**
